Letterbook.Web/Pages/Profile.cshtml

```diff
@@ -11,8 +11,11 @@
         <span>@Model.Handle</span>
     </h1>
     
-    <a href="/@@@Model.BareHandle/edit">Edit</a>
-    
+    @if (User.Identity?.IsAuthenticated == true)
+    {
+        <a asp-page="ProfileEdit" asp-route-handle="@Model.BareHandle">Edit</a>
+    }
+
```

Comment:
Note the conditional rendering, and the preferred way to construct links to other pages